src/api/fleet_addon_config.rs

```diff
@@ -239,6 +244,61 @@ pub struct FeatureGates {
 
     /// Enables experimental Helm operations support.
     pub experimental_helm_ops: bool,
+
+    // Enables syncing of feature gates to a ConfigMap.
+    pub config_map: Option<FeaturesConfigMap>,
+}
+
+impl FeatureGates {
+    /// Returns true if a ConfigMap reference is defined.
+    pub(crate) fn has_config_map_ref(&self) -> bool {
+        self.config_map.as_ref().and_then(|c | c.reference.as_ref()).is_some()
+    }
+
+    /// Syncs the `fleet` data key on the ConfigMap with FeatureGates
+    pub(crate) fn sync_configmap(&self, config_map: &mut ConfigMap) -> Result<(), Error> {
+        let mut empty_data: BTreeMap<String,String> = BTreeMap::new();
+        let data = config_map.data.as_mut().unwrap_or(&mut empty_data);
+        match data.get("fleet") {
+            Some(fleet_data) => {
+                data.insert(String::from("fleet"), self.merge_features(Some(fleet_data))?);
+            }
+            None => {
+                data.insert(String::from("fleet"), self.merge_features(None)?);
+            }
+        }
+        config_map.data = Some(data.clone());
+        Ok(())
+    }
+
+    /// Merge the feature gates environment variables with a provided optional input.
+    fn merge_features(&self, input: Option<&String>) -> Result<String, Error>  {
+        let mut extra_env_map: BTreeMap<String, String> = BTreeMap::new();
+        let mut values = FleetChartValues {
+            ..Default::default()
+        };
+
+        if let Some(input) = input {
+            values = serde_yaml::from_str(input).unwrap_or_default();
```

Comment:
```suggestion
            values = serde_yaml::from_str(input).map_err(?;
```
This may be a source of errors, if the configmap is malformed by a user intervention.